compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -655,3 +664,12 @@ fn promoted_mir(
 
     tcx.arena.alloc(promoted)
 }
+
+fn replace_base<'tcx>(place: &mut Place<'tcx>, new_base: Place<'tcx>, tcx: TyCtxt<'tcx>) {
```

Comment:
This method should be `*place = new_base.project_deeper(&place.projection[..], tcx)`.

Reply:
done in https://github.com/rust-lang/rust/pull/108590/commits/4f9e5272d392324d2aa6ff440a6dcfebd62a021c